src/BenchmarkDotNet.Core/Toolchains/InProcess/InProcessToolchain.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using BenchmarkDotNet.Characteristics;
+using BenchmarkDotNet.Loggers;
+using BenchmarkDotNet.Running;
+using JetBrains.Annotations;
+
+namespace BenchmarkDotNet.Toolchains.InProcess
+{
+    /// <summary>
+    ///     A toolchain to run the benchmarks in-process.
+    /// </summary>
+    /// <seealso cref="IToolchain" />
+    [PublicAPI]
+    public sealed class InProcessToolchain : IToolchain
+    {
+        /// <summary>The default toolchain instance.</summary>
+        public static readonly IToolchain Instance = new InProcessToolchain(true);
+
+        /// <summary>The toolchain instance without output logging.</summary>
+        public static readonly IToolchain DontLogOutput = new InProcessToolchain(false);
+
+        /// <summary>Initializes a new instance of the <see cref="InProcessToolchain" /> class.</summary>
+        /// <param name="logOutput"><c>true</c> if the output should be logged.</param>
+        public InProcessToolchain(bool logOutput) : this(
+            InProcessExecutor.DefaultTimeout,
+            BenchmarkActionCodegen.ReflectionEmit,
+            logOutput)
+        {
+        }
+
+        /// <summary>Initializes a new instance of the <see cref="InProcessToolchain" /> class.</summary>
+        /// <param name="timeout">Timeout for the run.</param>
+        /// <param name="codegenMode">Describes how benchmark action code is generated.</param>
+        /// <param name="logOutput"><c>true</c> if the output should be logged.</param>
+        public InProcessToolchain(TimeSpan timeout, BenchmarkActionCodegen codegenMode, bool logOutput)
+        {
+            Generator = new InProcessGenerator();
+            Builder = new InProcessBuilder();
+            Executor = new InProcessExecutor(timeout, codegenMode, logOutput);
+        }
+
+        /// <summary>Determines whether the specified benchmark is supported.</summary>
+        /// <param name="benchmark">The benchmark.</param>
+        /// <param name="logger">The logger.</param>
+        /// <param name="resolver">The resolver.</param>
+        /// <returns><c>true</c> if the benchmark can be run with the toolchain.</returns>
+        public bool IsSupported(Benchmark benchmark, ILogger logger, IResolver resolver) => true;
```

Comment:
I believe that there are few scenarios that you can not support. For example: architecture/jit different than host/current